### PR TITLE
[wasm][debugger] Correct the endian swapping and string implementation

### PR DESCRIPTION
**Pull request by lewing**
A limited change fix WriteString and only byte swap when needed.

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 1]
```diff
@@ -409,105 +409,71 @@ public override string ReadString()
...
-        }
     }
 
     internal class MonoBinaryWriter : BinaryWriter
     {
         public MonoBinaryWriter(Stream stream) : base(stream) {}
+
         public void WriteString(string val)
         {
+            var bytes = Encoding.UTF8.GetBytes(val);
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:462]: did you mean to use `bytes` for `Write(..)`?

Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:440]: 🤦

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 2]
```diff
@@ -407,108 +408,50 @@ public override string ReadString()
...
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
-            int ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
+            Span<byte> data = stackalloc byte[Unsafe.SizeOf<T>()];
+            T ret = default;
+            Read(data);
+            SwapForBE(new Span<byte>(Unsafe.AsPointer(ref ret), data.Length), data);
```
Comment (marek-safar) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:429]: What about 
```c#
if (BitConverter.IsLittleEndian)
	data.Reverse();
```
instead?

Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs]: I'm fine with that.

**Review by thaystg — APPROVED** [review]

**Pull request merged**